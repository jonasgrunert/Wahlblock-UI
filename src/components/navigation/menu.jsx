import { Icon, Menu, MenuList, Title } from 'bloomer';
import * as React from 'react';
import { push as SideBar } from 'react-burger-menu';
import { decorator as reduxBurgerMenu } from 'redux-burger-menu';

import burgerMenuStyles from '../../config/burgerMenuStyles';
import menuLinks from '../../config/menuLink';
import NavLinkWrapper from './navLinkWrapper';

const Links = menuLinks.map(link => (
  <NavLinkWrapper link={link.link} name={link.name} />
));

const AdminMenu = () => (
  <Menu>
    <MenuList>
      { Links }
    </MenuList>
  </Menu>
);


const SideMenu = () => (
  <SideBar
    isOpen={false}
    pageWrapId="main"
    outerContainerId="outer-container"
    styles={burgerMenuStyles}
    customBurgerIcon={
      <div>
        <Icon isDisplay="inline" isSize="medium" className="fa fa-bars fa-2x" />
        <Title isDisplay="inline" isSize={3}>&nbsp;Wahlblock</Title>
      </div>
    }
  >
    <AdminMenu />
  </SideBar>
);

const MainMenu = reduxBurgerMenu(SideMenu);
export default MainMenu;
