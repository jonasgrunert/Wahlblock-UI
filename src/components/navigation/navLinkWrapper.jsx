import { MenuLink } from 'bloomer';
import * as React from 'react';
import { connect } from 'react-redux';
import { push } from 'react-router-redux';
import { action as toggleMenu } from 'redux-burger-menu';
import PropTypes from 'prop-types';

// Button displayed and rendered as active
export const NavLinkButton = props => (
  <li>
    <MenuLink
      onClick={props.onLink}
      isActive={props.location === props.link}
      hasTextColor={props.location === props.link ? 'white' : 'grey'}
    >
      {props.name}
    </MenuLink>
  </li>
);

NavLinkButton.propTypes = {
  name: PropTypes.string.isRequired,
  onLink: PropTypes.func.isRequired,
  location: PropTypes.string.isRequired,
  link: PropTypes.string.isRequired,
};

// map redux state to props of NavLinkButton
const mapStateToProps = state => ({
  // url
  location: state.routerReducer.location.pathname,
});

// map dispatch actions to props of NavLinkButton
const mapDispatchToProps = (dispatch, ownProps) => ({
  // closing and opening sidemenu and navigating
  onLink: (event) => {
    event.preventDefault();
    dispatch(toggleMenu(true));
    dispatch(toggleMenu(false));
    dispatch(push(ownProps.link));
  },
});

// creating higher order component
const NavLinkWrapper = connect(mapStateToProps, mapDispatchToProps)(NavLinkButton);
export default NavLinkWrapper;
