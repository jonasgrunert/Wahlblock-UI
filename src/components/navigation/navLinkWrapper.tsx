import { MenuLink } from "bloomer";
import * as React from "react";
import { connect } from "react-redux";
import { NavLink, withRouter } from "react-router-dom";
import { push, RouterState } from "react-router-redux";
import { Dispatch } from "redux";
import {action as toggleMenu, IStateReduxBurgerMenu} from "redux-burger-menu";

import { IStore } from "../../index";

interface IOwnProps {
  link: string;
  name: string;
}

interface IStateProps {
  location: Location;
}

interface IDispatchProps {
  onLink: () => any;
}

interface IProps {
  link: string;
  name: string;
  onLink: () => any;
  // location: RouterState;
}

const NavLinkButton = (props: IOwnProps & IDispatchProps) =>
  <li>
    <MenuLink onClick={props.onLink}>
        {props.name}
    </MenuLink>
  </li>;

const mapStateToProps = (state: IStore) => ({
  location: state.router,
});

const mapDispatchToProps = (dispatch: Dispatch<any>, ownProps: IOwnProps) => ({
  onLink: () => {
    dispatch(toggleMenu(false));
    dispatch(push(ownProps.link));
  },
});


export const NavLinkWrapper = connect(
    null,
    mapDispatchToProps,
  )(NavLinkButton);

// export default withRouter(NavLinkWrapper);
