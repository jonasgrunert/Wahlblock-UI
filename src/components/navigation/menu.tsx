import { Icon, Image, Menu, MenuLink, MenuList, Subtitle, Title } from "bloomer";
import * as React from "react";
import { push as SideBar } from "react-burger-menu";
import {decorator as reduxBurgerMenu} from "redux-burger-menu";

import { burgerMenuStyles } from "../../config/burgerMenuStyles";
import { menuLinks } from "../../config/menuLink";
import { NavLinkWrapper } from "./navLinkWrapper";

const Links = menuLinks.map((link) => {
  return (
    <NavLinkWrapper link={link.link} name={link.name} />
  );
});

const AdminMenu = () =>
  <Menu>
    <MenuList>
      { Links }
    </MenuList>
  </Menu>;


export const SideMenu = () =>
  <SideBar
    pageWrapId={"main"}
    outerContainerId={"outer-container"}
    styles={burgerMenuStyles}
    customBurgerIcon={
      <div>
        <Icon isDisplay="inline" isSize="medium" className="fa fa-bars fa-2x" />
        <Title isDisplay="inline" isSize={3}>&nbsp;Wahlblock</Title>
      </div>
    }
  >
    <AdminMenu />
  </SideBar>;

export default reduxBurgerMenu(SideMenu);
