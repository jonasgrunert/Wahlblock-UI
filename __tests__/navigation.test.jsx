import * as React from 'react';
import { Links, SubMenus, AdminMenu, SideMenu } from '../src/components/navigation/menu';
import { NavLinkButton } from '../src/components/navigation/navLinkWrapper';

const menuConfig = [
  {
    exact: true,
    link: '/home',
    name: 'Home',
  },
  {
    exact: false,
    link: '/default',
    name: 'Home',
  },
];

const menusConfig = [
  {
    title: 'default',
    id: 0,
    base: 'default',
    links: menuConfig,
  },
];

const base = 'local';

const button = 'navigation';

describe('rendering navigation', () => {
  it('render navigationlink active', () => {
    const navlink = shallow(<NavLinkButton name={button} onLink={jest.fn()} location={button} link={button} />);
    expect(navlink).toMatchSnapshot();
  });
  it('render navigationlink active', () => {
    const navlink = shallow(<NavLinkButton name={button} onLink={jest.fn()} location="button" link={button} />);
    expect(navlink).toMatchSnapshot();
  });
  it('render links', () => {
    const links = Links(menuConfig, base);
    expect(links).toMatchSnapshot();
  });
  it('render submenus', () => {
    const submenus = SubMenus(menusConfig);
    expect(submenus).toMatchSnapshot();
  });
  it('render adminmenu', () => {
    const adminmenu = shallow(<AdminMenu routesConfig={menusConfig} />);
    expect(adminmenu).toMatchSnapshot();
  });
  it('render sidemenu', () => {
    const sidemenu = shallow(<SideMenu routesConfig={menusConfig} title={base} />);
    expect(sidemenu).toMatchSnapshot();
  });
});
