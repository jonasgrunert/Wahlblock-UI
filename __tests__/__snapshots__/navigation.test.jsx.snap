// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering navigation render adminmenu 1`] = `
<withHelpersModifiers(Menu)>
  <withHelpersModifiers(MenuList)>
    <withHelpersModifiers(Container)
      isFluid={true}
      key="default"
      style={
        Object {
          "margin": "1em",
        }
      }
    >
      <withHelpersModifiers(MenuLabel)>
        default
      </withHelpersModifiers(MenuLabel)>
      <withHelpersModifiers(MenuList)>
        <Connect(NavLinkButton)
          key="/home"
          link="/default//home"
          name="Home"
        />
        <Connect(NavLinkButton)
          key="/default"
          link="/default//default"
          name="Home"
        />
      </withHelpersModifiers(MenuList)>
    </withHelpersModifiers(Container)>
  </withHelpersModifiers(MenuList)>
</withHelpersModifiers(Menu)>
`;

exports[`rendering navigation render links 1`] = `
Array [
  <Connect(NavLinkButton)
    link="/local//home"
    name="Home"
  />,
  <Connect(NavLinkButton)
    link="/local//default"
    name="Home"
  />,
]
`;

exports[`rendering navigation render navigationlink active 1`] = `
<li>
  <withHelpersModifiers(MenuLink)
    hasTextColor="white"
    isActive={true}
    onClick={[MockFunction]}
  >
    navigation
  </withHelpersModifiers(MenuLink)>
</li>
`;

exports[`rendering navigation render navigationlink active 2`] = `
<li>
  <withHelpersModifiers(MenuLink)
    hasTextColor="grey"
    isActive={false}
    onClick={[MockFunction]}
  >
    navigation
  </withHelpersModifiers(MenuLink)>
</li>
`;

exports[`rendering navigation render sidemenu 1`] = `
<Menu
  bodyClassName=""
  burgerBarClassName=""
  burgerButtonClassName=""
  className=""
  crossButtonClassName=""
  crossClassName=""
  customBurgerIcon={
    <div>
      <withHelpersModifiers(Icon)
        className="fa fa-bars fa-2x"
        isDisplay="inline"
        isSize="medium"
      />
      <withHelpersModifiers(Title)
        isDisplay="inline"
        isSize={3}
      >
        Â 
        local
      </withHelpersModifiers(Title)>
    </div>
  }
  disableCloseOnEsc={false}
  id=""
  isOpen={false}
  itemListClassName=""
  menuClassName=""
  morphShapeClassName=""
  noOverlay={false}
  onStateChange={[Function]}
  outerContainerId="outer-container"
  overlayClassName=""
  pageWrapId="main"
  styles={
    Object {
      "bmBurgerBars": Object {
        "background": "#373a47",
      },
      "bmBurgerButton": Object {
        "height": "30px",
        "left": "36px",
        "position": "fixed",
        "top": "20px",
        "width": "200px",
      },
      "bmCross": Object {
        "background": "#bdc3c7",
      },
      "bmCrossButton": Object {
        "height": "24px",
        "width": "24px",
      },
      "bmItemList": Object {
        "color": "#b8b7ad",
        "paddingTop": "2em",
      },
      "bmMenu": Object {
        "background": "#ffffff",
        "fontSize": "1.15em",
      },
      "bmMorphShape": Object {
        "fill": "#373a47",
      },
      "bmOverlay": Object {
        "background": "rgba(0, 0, 0, 0.3)",
      },
    }
  }
  width={300}
>
  <AdminMenu
    routesConfig={
      Array [
        Object {
          "base": "default",
          "id": 0,
          "links": Array [
            Object {
              "exact": true,
              "link": "/home",
              "name": "Home",
            },
            Object {
              "exact": false,
              "link": "/default",
              "name": "Home",
            },
          ],
          "title": "default",
        },
      ]
    }
  />
</Menu>
`;

exports[`rendering navigation render submenus 1`] = `
Array [
  <withHelpersModifiers(Container)
    isFluid={true}
    style={
      Object {
        "margin": "1em",
      }
    }
  >
    <withHelpersModifiers(MenuLabel)>
      default
    </withHelpersModifiers(MenuLabel)>
    <withHelpersModifiers(MenuList)>
      <Connect(NavLinkButton)
        link="/default//home"
        name="Home"
      />
      <Connect(NavLinkButton)
        link="/default//default"
        name="Home"
      />
    </withHelpersModifiers(MenuList)>
  </withHelpersModifiers(Container)>,
]
`;
